packages/truffle-compile/compilerSupplier/loadingStrategies/VersionRange.js

```diff
@@ -0,0 +1,100 @@
+const fs = require("fs");
+const ora = require("ora");
+const semver = require("semver");
+const LoadingStrategy = require("./LoadingStrategy");
+
+class VersionRange extends LoadingStrategy {
+  load(versionRange) {
+    const rangeIsSingleVersion = semver.valid(versionRange);
+    return rangeIsSingleVersion
+      ? this.getSolcBySingleVersion(versionRange)
+      : this.getSolcByVersionRange(versionRange);
```

Comment:
Is the reason for this distinction because of performance?

I vote for a comment here to say why we treat the special case specially.

Reply:
Yes, for the case where a single version is specified we will go straight to the cache to check if it is there. I'll add a comment!